src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -66,7 +66,7 @@
         private static final int M2 = 0x33333333;
         private static final int M4 = 0x0f0f0f0f;
 
-        int bitCount(int x) {
+        static int bitCount(int x) {
```

Comment:
What's the benefit of making it static? Just asking. `static` is fine! It makes clear that we do not depend on any (mutable) state :-) 

We save the implicit `this`. Are there more benefits? I've read somewhere that "Memory is allocated for a method when the method is invoked. (And it doesn't matter whether the method is a static or instance method.". The bytecode for non-static methods is allocated only once and re-used by every instance.